src/hyperion/external_interaction/callbacks/ispyb_callback_base.py

```diff
@@ -42,44 +42,45 @@ def descriptor(self, doc: dict):
         self.descriptors[doc["uid"]] = doc
 
     def start(self, doc: dict):
+        LOGGER.info(f"ISPyB handler received start document {doc}.")
         if self.uid_to_finalize_on is None:
             self.uid_to_finalize_on = doc.get("uid")
 
     def event(self, doc: dict):
         """Subclasses should extend this to add a call to set_dcig_tag from
         hyperion.log"""
 
-        LOGGER.debug("ISPyB handler received event document.")
+        LOGGER.debug(f"ISPyB handler received event document {doc}.")
         assert isinstance(
             self.ispyb, StoreInIspyb
         ), "ISPyB deposition can't be initialised!"
         event_descriptor = self.descriptors[doc["descriptor"]]
 
         if event_descriptor.get("name") == ISPYB_HARDWARE_READ_PLAN:
-            self.params.hyperion_params.ispyb_params.undulator_gap = doc["data"][
-                "undulator_gap"
-            ]
-            self.params.hyperion_params.ispyb_params.synchrotron_mode = doc["data"][
-                "synchrotron_machine_status_synchrotron_mode"
-            ]
-            self.params.hyperion_params.ispyb_params.slit_gap_size_x = doc["data"][
-                "s4_slit_gaps_xgap"
-            ]
-            self.params.hyperion_params.ispyb_params.slit_gap_size_y = doc["data"][
-                "s4_slit_gaps_ygap"
-            ]
+            self.params.hyperion_params.ispyb_params.undulator_gap = doc["data"].get(
+                "undulator_gap", 0.0
+            )
+            self.params.hyperion_params.ispyb_params.synchrotron_mode = doc["data"].get(
+                "synchrotron_machine_status_synchrotron_mode", None
+            )
+            self.params.hyperion_params.ispyb_params.slit_gap_size_x = doc["data"].get(
+                "s4_slit_gaps_xgap", 0.0
+            )
+            self.params.hyperion_params.ispyb_params.slit_gap_size_y = doc["data"].get(
+                "s4_slit_gaps_ygap", 0.0
+            )
 
         if event_descriptor.get("name") == ISPYB_TRANSMISSION_FLUX_READ_PLAN:
             self.params.hyperion_params.ispyb_params.transmission_fraction = doc[
                 "data"
-            ]["attenuator_actual_transmission"]
-            self.params.hyperion_params.ispyb_params.flux = doc["data"][
-                "flux_flux_reading"
-            ]
+            ].get("attenuator_actual_transmission", 0.0)
+            self.params.hyperion_params.ispyb_params.flux = doc["data"].get(
+                "flux_flux_reading", 0.0
+            )
 
-            LOGGER.info("Creating ispyb entry.")
-            self.ispyb_ids = self.ispyb.begin_deposition()
-            LOGGER.info(f"Recieved ISPYB IDs: {self.ispyb_ids}")
+        LOGGER.info("Creating ispyb entry.")
```

Comment:
How come this part is no longer within the if statement?

Reply:
If I recally correctly, VMXm only uses the `READ_HARDWARE` plan so they need to create the ispyb entry on that event - otherwise they won't get an entry at all. For `i03`, we need to check whether this is right. 

It feels like a bit of a hack anyway to have this ispyb stuff listening on a specific type of message - but I think David or Dom have been doing some work to pull the ispyb interactions into an ophyd device? If that's the case then this may go away.